### PR TITLE
Change DescribeMutableState API to proto

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
`DescribeMutableState` API was changed to proto from JSON.

**Why?**
For consistency, performance, and to get rid of ugly code.

**How did you test it?**
Run tests and `admin workflow` commands manually.

**Potential risks**
`admin workflow` commands have breaking change.

### CODE REVIEW

File: client/admin/retryableClient.go [hunk 1]
```diff
@@ -114,16 +114,16 @@ func (c *retryableClient) CloseShard(
...
 }
 
-func (c *retryableClient) DescribeWorkflowExecution(
+func (c *retryableClient) DescribeMutableState(
 	ctx context.Context,
-	request *adminservice.DescribeWorkflowExecutionRequest,
+	request *adminservice.DescribeMutableStateRequest,
 	opts ...grpc.CallOption,
-) (*adminservice.DescribeWorkflowExecutionResponse, error) {
+) (*adminservice.DescribeMutableStateResponse, error) {
```
Comment (wxing1292) [inline client/admin/retryableClient.go:121]: breaking change?

File: service/frontend/adminHandler.go [hunk 2]
```diff
@@ -214,21 +214,21 @@ func (adh *AdminHandler) DescribeWorkflowExecution(ctx context.Context, request
...
 		ShardId:              shardIDStr,
 		HistoryAddr:          historyAddr,
-		DatabaseMutableState: resp2.GetDatabaseMutableState(),
-		CacheMutableState:    resp2.GetCacheMutableState(),
-		TreeId:               resp2.GetTreeId(),
-		BranchId:             resp2.GetBranchId(),
+		DatabaseMutableState: historyResponse.GetDatabaseMutableState(),
+		CacheMutableState:    historyResponse.GetCacheMutableState(),
+		// TreeId:               historyResponse.GetTreeId(),
+		// BranchId:             historyResponse.GetBranchId(),
```
Comment (wxing1292) [inline service/frontend/adminHandler.go:231]: uncomment?

File: tools/cli/adminCommands.go [hunk 3]
```diff
@@ -122,80 +122,72 @@ func AdminShowWorkflow(c *cli.Context) {
...
+	// 	BranchID    string
+	// 	ShardID     string
+	// 	HistoryAddr string
+	// }{
+	// 	resp.GetTreeId(),
+	// 	resp.GetBranchId(),
+	// 	resp.GetShardId(),
+	// 	resp.GetHistoryAddr(),
+	// }
+	// prettyPrintJSONObject(output)
```
Comment (wxing1292) [inline tools/cli/adminCommands.go:158]: remove or uncomment

File: client/admin/retryableClient.go (hunk 1 above)
Comment (alexshtin) [reply client/admin/retryableClient.go:121]: Yes. Unfortunately I didn't have time to fix this before release. But this API is used by admin CLI only, which is ok to break. I will outline this in next release notes (one `tctl admin` command works works only with corresponding version of server).

File: proto/internal/temporal/server/api/adminservice/v1/service.proto [hunk 4]
```diff
@@ -29,7 +29,7 @@ import "temporal/server/api/adminservice/v1/request_response.proto";
 service AdminService {
 
     // DescribeWorkflowExecution returns information about the internal states of workflow execution.
-    rpc DescribeWorkflowExecution (DescribeWorkflowExecutionRequest) returns (DescribeWorkflowExecutionResponse) {
+    rpc DescribeMutableState (DescribeMutableStateRequest) returns (DescribeMutableStateResponse) {
```
Comment (alexshtin) [inline proto/internal/temporal/server/api/adminservice/v1/service.proto:32]: I decided to rename API to match `historyservice` and CLI code.

File: service/history/mutableStateBuilder.go [hunk 5]
```diff
@@ -260,22 +260,20 @@ func newMutableStateBuilderWithVersionHistories(
...
-	state.SignalInfos = e.pendingSignalInfoIDs
-	state.SignalRequestedIds = convertStringSetToSlice(e.pendingSignalRequestedIDs)
-	state.ExecutionInfo = e.executionInfo
-	state.ExecutionState = e.executionState
-	state.NextEventId = e.nextEventID
-	state.BufferedEvents = e.bufferedEvents
-	state.Checksum = e.checksum
-
-	return state
+func (e *mutableStateBuilder) CopyToProto() *persistencespb.WorkflowMutableState {
```
Comment (samarabbas) [inline service/history/mutableStateBuilder.go:263]: Why is it called copy?  This is more like `MutableAsProto`

Comment (samarabbas) [reply service/history/mutableStateBuilder.go:263]: By the way, should we actually considering creating a copy (clone)?

**Review by samarabbas — APPROVED** [review]
Overall looks good.  One minor comment.

File: service/history/mutableStateBuilder.go (hunk 5 above)
Comment (alexshtin) [reply service/history/mutableStateBuilder.go]: Agree, word "copy" might be confusing. Renamed it to `ToProto`. Actual copy is not needed there, as this data get serialized right away (=copy).

**Pull request merged**